### PR TITLE
multigrid tools: Remove template on sparsity pattern type.

### PR DESCRIPTION
**Pull request by drwells**
Another followup to #14396: part 9/11.

### CODE REVIEW

File: source/multigrid/mg_tools.cc [hunk 1]
```diff
@@ -780,7 +767,16 @@ namespace MGTools
...
     std::vector<types::global_dof_index> dofs_on_other_cell(total_dofs);
-    Table<2, bool>                       support_on_face(total_dofs,
+    std::vector<types::global_dof_index> rows;
+    std::vector<types::global_dof_index> cols;
+
+    auto append_pair = [&](const types::global_dof_index i,
+                           const types::global_dof_index j) {
+      rows.push_back(i);
+      cols.push_back(j);
+    };
```
Comment (bangerth) [inline source/multigrid/mg_tools.cc:777]: Same concern as in #14434.

**Review by bangerth — COMMENTED** [review]
Same concern as in #14434.

**Comment by drwells** [comment]
Lets do #14441 first and then redo this one with the new API.

**Comment by drwells** [comment]
Updated!

File: source/multigrid/mg_tools.cc [hunk 2]
```diff
@@ -1028,15 +1029,25 @@ namespace MGTools
...
-                sparsity.add(dofs_on_this_cell[i], dofs_on_this_cell[j]);
+            {
+              cols.clear();
+              for (unsigned int j = 0; j < dofs_per_cell; ++j)
+                if (mg_constrained_dofs.is_interface_matrix_entry(
+                      level, dofs_on_this_cell[i], dofs_on_this_cell[j]))
+                  cols.push_back(dofs_on_this_cell[j]);
+              sparsity.add_row_entries(dofs_on_this_cell[i],
+                                       make_array_view(cols),
+                                       true);
```
Comment (kronbichler) [inline source/multigrid/mg_tools.cc:1049]: Would you mind making this consistent with the other functions and clear after adding entries,
```suggestion
              for (unsigned int j = 0; j < dofs_per_cell; ++j)
                if (mg_constrained_dofs.is_interface_matrix_entry(
                      level, dofs_on_this_cell[i], dofs_on_this_cell[j]))
                  cols.push_back(dofs_on_this_cell[j]);
              sparsity.add_row_entries(dofs_on_this_cell[i],
                                       make_array_view(cols),
                                       true);
              cols.clear();
```

**Review by kronbichler — APPROVED** [review]
nice!

**Pull request merged**